TG Armor Rebalance

Fox-McCloud opened the pull request:
TG's armor balance changes. 

Issue attempting to be addressed: Keep armor viable, but overall make SS13 a bit more lethal--also reduces the dominance that range weapons have over their melee counterparts. Makes combat more scary, but also more mobile.

This also addresses some things brought up by the armor refactor a long time ago--when armor was made from being RNG based to guaranteed. While, statistically, the armor block was the same, after the rework, in actuality, armor became stronger, because it was guaranteed (ie: 10% damage reduction is generally more valuable than a 10% chance to block 100% damage).

Major changes are just listed in the changelog

@Aurorablade has expressed interesting is porting TG's hit_reaction system, which this should pair well with.

Let the salt flow.

:cl: Fox McCloud
tweak: Most melee and laser armor values halved or reduced; some bullet armor reduced slightly
tweak: Slowdown on most armor and spacesuits reduced by 1 (chaplain armor is still slow though)
tweak: Mining armor caps out at 60 melee resist as opposed to 80
tweak: Ling melee armor reduced; bullet and laser values buffed a bit and slowdown removed
tweak: Chameleon jumpsuit has a little bit of armor
tweak: Detective's forensic jacket now has identical armor to the native detective jacket
tweak: Armor penetration added to energy swords, chainsaws, scythes, and spears
/:cl:

tigercat2000 commented:
could you please adjust the armour values on the bay RIGsuits to match everything with this? https://github.com/ParadiseSS13/Paradise/tree/master/code/modules/clothing/spacesuits/rig/suits

FreeStylaLT commented:
Do you think this warrants an increase in the prices of swords and chainsaws?

FreeStylaLT commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -82,19 +82,10 @@
 	item_state = "helmet"
 	flags = HEADCOVERSEYES | HEADCOVERSMOUTH | HEADBANGPROTECT
-	armor = list(melee = 80, bullet = 10, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0)
+	armor = list(melee = 41, bullet = 15, laser = 5, energy = 5, bomb = 5, bio = 2, rad = 0)
```
41?

Fox-McCloud replied:
It can't be the answer to everything.

theColdflame commented:
:-1: as the PR stands, far too severe in my book after looking at some of the numbers.

stuns are already king, reducing the already questionable combat effectiveness of armor seems like an odd change.

this is coming from someone who largely plays sec, but I'm not sure of the actual reasoning behind this

Ssirius commented:
SS13 More lethal? Sign me up :+1:

Fox-McCloud commented:
@tigercat2000 

Sure, if it looks like this is going to gain traction, I'll change those, too.

Spacemanspark commented:
I'm unsure as to how I feel about this.

tigercat2000 commented:
Other than that-

I'd like to give this a try. See how it is, before judging "oh god everyone's going to die RIP fuck everyone balance patches never get reverted."
It's hard to judge armor values without seeing them in live action.

theColdflame commented:
realistically, this PR won't get reverted if merged

regardless, I'm not entirely against this change, but I'd rather have it discussed beyond '/tg/ has it' given how heavy of a nerf it is.

Dumbdumn5 commented:
I'd give it a shot, and what do you mean "hit reactions"?

Fox-McCloud commented:
@Dumbdumn5 

A system whereby we can code things to happen when someone, that has that item, gets hit.

For example, we could make it so that when you punch someone who is wearing a clown outfit, they will honk.

Could also make armor has 100% armor for everything (ie, blocks shots), but each time it gets hit, it reduces its charge by one...with it slowly recharging over time.

Orrrr if someone is holding a grenade and you shoot them, it has a chance of triggering the grenade.

Stuff like that.

Riot Shields use it extensively in combination with armor penetration.

TheDZD commented on code/game/objects/items/weapons/melee/energy.dm:
```diff
@@ -63,6 +63,7 @@
 	w_class_on = 5
 	hitsound = "swing_hit"
 	flags = CONDUCT | NOSHIELD
+	armour_penetration = 100
```
150 force, complete armor penetration. The pain is real.

TheDZD commented on code/game/objects/items/weapons/staff.dm:
```diff
@@ -9,6 +9,7 @@
 	throw_range = 5
 	w_class = 2.0
 	flags = NOSHIELD
+	armour_penetration = 100
```
Magic (but it only has 3 force).

TheDZD commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -203,19 +194,27 @@ obj/item/clothing/head/blob
 	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES
 
-/obj/item/clothing/head/helmet/knight/blue
+/obj/item/clothing/head/helmet/riot/knight
```
Might want to move this, and the rest of the knight helmets to just below the riot helmet, that way it's easier to immediately see what gets inherited.

TheDZD commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -21,15 +21,15 @@
 	icon_state = "jackboots"
 	item_state = "jackboots"
-	armor = list(melee = 50, bullet = 50, laser = 50, energy = 25, bomb = 50, bio = 10, rad = 0)
+	armor = list(melee = 25, bullet = 25, laser = 25, energy = 25, bomb = 50, bio = 10, rad = 0)
```
Huh, do shoe armor values actually do anything?

Fox-McCloud replied:
They do, indeed.

Ar3nn replied:
I'd guess if you hit their foot (maybe leg, since they're boots?) it'd be a factor

TheDZD commented on code/modules/clothing/spacesuits/alien.dm:
```diff
@@ -76,15 +76,13 @@
 /obj/item/clothing/suit/space/vox
 	w_class = 3
 	allowed = list(/obj/item/weapon/gun,/obj/item/ammo_box,/obj/item/ammo_casing,/obj/item/weapon/melee/baton,/obj/item/weapon/melee/energy/sword/saber,/obj/item/weapon/restraints/handcuffs,/obj/item/weapon/tank)
-	slowdown = 2
```
Super speed Vox raiders.

KasparoVy replied:
Gotta go fast! Makes sense since their goal isn't to fight but to grab stuff and split.

Spacemanspark commented:
Mjolnir armor when.

TheDZD commented on code/modules/clothing/spacesuits/rig.dm:
```diff
@@ -420,7 +417,7 @@
 	item_color = "wiz"
 	unacidable = 1 //No longer shall our kind be foiled by lone chemists with spray bottles!
-	armor = list(melee = 40, bullet = 20, laser = 20,energy = 20, bomb = 35, bio = 100, rad = 60)
+	armor = list(melee = 40, bullet = 40, laser = 40, energy = 20, bomb = 35, bio = 100, rad = 50)
```
Maybe one day wizards will actually take this.

TheDZD commented on code/modules/clothing/spacesuits/syndi.dm:
```diff
@@ -5,7 +5,7 @@
 	item_state = "syndicate"
 	desc = "Has a tag: Totally not property of an enemy corporation, honest."
-	armor = list(melee = 60, bullet = 50, laser = 30,energy = 15, bomb = 30, bio = 30, rad = 30)
+	armor = list(melee = 40, bullet = 50, laser = 30,energy = 15, bomb = 30, bio = 30, rad = 30)
```
I like how you can just tell where coders at some point just copy+pasted shit, because of small inconsistencies being copied over and over.

KasparoVy replied:
Could we put a space between energy and that comma there?

TheDZD commented on code/modules/clothing/suits/jobs.dm:
```diff
@@ -149,7 +149,7 @@
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
 	allowed = list(/obj/item/weapon/tank/emergency_oxygen, /obj/item/device/flashlight,/obj/item/weapon/gun/energy,/obj/item/weapon/gun/projectile,/obj/item/ammo_box,/obj/item/ammo_casing,/obj/item/weapon/melee/baton,/obj/item/weapon/restraints/handcuffs,/obj/item/device/detective_scanner,/obj/item/device/taperecorder)
-	armor = list(melee = 10, bullet = 10, laser = 15, energy = 10, bomb = 0, bio = 0, rad = 0)
+	armor = list(melee = 10, bullet = 10, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0)
```
Might want to make this just be a reskinned detective suit, so it's just a cosmetic choice.

KasparoVy commented:
I'd love to try it.

Vivalas commented:
I'm all for making SS13 more lethal, but I think the way to do that is to make weapons more lethal, not armor less useful. Armor is already an afterthought to most people when engaging in combat, with dubious usefulness. Most people would agree weapon choice and stuns are a lot more important in combat than armor. As sec I always avoid riot armor like the plague because the immense slowdown is not worth the armor boost, especially in a game like this, where mobility has a HUGE affect on outcome. Of course, I might consider riot armor from now on, but it's still a questionable tradeoff.

Making armor less useful does not make the game more lethal for the most people, as the vast majority of the station goes into combat without any armor. Making weapons more lethal affects everyone. If you buffed both armor AND weapons drastically, it would make armor a much more important thing to consider as combat without it would be more lethal. But this is just my opinion, I would like to see this play out.

Dumbdumn5 commented:
Oh, cool.

FlattestGuitar commented:
One thing I don't like here is the mining hardsuits. Almost every shift there's at least one casualty on the asteroid and with this nerf  we're just punishing the newer players who won't be able to beat a Goliath with armor values this low, while more experienced ones will get the suit to the 60 armor and still have a very strong suit. 

I'd leave its current melee value where it is, only reducing the possible max upgrade. 

Other than that 👍.

Fox-McCloud commented:
Alright.

- Tweaked Bay Rig armor values 
- Removed some stats from unused/unmapped species-only space suits
- Made forensic technician jackets a subtype of the detective's normal jacket
- corrected a lot of armor spacing consistency
- Gave the chameleon jumpsuit a bit of armor

Ar3nn commented:
I agree with Flattest about the mining suits, 80 armor sounds like a lot, but Goliaths hit for a FUCKTON and any emergency that a miner would react to, they realistically wouldn't have farmed all the plates yet, anyway. I consider it a non-issue, but would prefer it left alone.
